### PR TITLE
added filtering and sorting

### PR DESCRIPTION
**Pull request by Wobin**
added wallet details

### CODE REVIEW

**Comment by Wobin** [comment]
I'm merging from a branch this time, so maybe it'll fix the whole '22 commits' thing when I discard changes on my main branch after this is merged

File: app/routes/armoury/$character/exchange.tsx [hunk 1]
```diff
@@ -25,104 +46,218 @@ export async function loader({ request, params }: LoaderArgs) {
...
-					</div>
-					{offer.item.description.overrides.perks && offer.item.description.overrides.perks.length > 0 ? (
+		<>
+			<div className="grid w-full grow grid-cols-2 gap-4 bg-neutral-200 p-4 shadow-inner">
+				{offers.map((offer) => (
+					<div
+						key={offer.id}
+						className={classnames(
+							"border-l-3 from-1% relative border-2 border-neutral-400 bg-white bg-gradient-to-r shadow",
+							rarityBorder[String(offer.rarity) ?? "0"]
```
Comment (danreeves) [inline app/routes/armoury/$character/exchange.tsx:180]:
```suggestion
							rarityBorder[offer.rarity]
```

File: app/routes/armoury/$character/exchange.tsx [hunk 2]
```diff
@@ -25,104 +46,218 @@ export async function loader({ request, params }: LoaderArgs) {
...
+								<span></span>
+							</div>
+						) : null}
+
+						<img
+							alt=""
+							loading="lazy"
+							className="max-w-1/2 pointer-events-none absolute right-0 top-0 h-full"
+							src={`https://img.darkti.de/pngs/${offer.shopitem.preview_image}.png`}
+						/>
```
Comment (danreeves) [inline app/routes/armoury/$character/exchange.tsx:235]: check out the Img component I added to the components directory, used in the inventory

**Review by danreeves — APPROVED** [review]
nice, i like the zod schema transform!

**Pull request merged**